osmo_jupyter/plot.py

```diff
@@ -0,0 +1,277 @@
+import plotly.graph_objs as go
+
+
+COLOR_CHANNELS = ['r', 'g', 'b']
+
+# Axis # to use for each color when colors are being plotted on separate axes
+AXES_BY_COLOR = {
+    'r': 2,
+    'g': 3,
+    'b': 4,
+}
+ANNOTATION_AXIS = 1
+SHARED_COLOR_AXIS = 2
+
+AXIS_SIDES_BY_COLOR = {
+    'r': 'left',
+    'g': 'right',
+    'b': 'right',
+}
+
+COLORS_BY_LETTER = {
+    'r': 'red',
+    'g': 'green',
+    'b': 'blue'
+}
+
+
+def subsample_for_plot(df, dots_to_plot=150):
+    ''' resample every nth row so that Plotting doesnt use so much CPU
+    '''
+    n = int(len(df) / dots_to_plot) or 1
+    return df.iloc[::n, :]
+
+
+def get_scatter(
+    y_values,
+    x_values=None,
+    dataset_name=None,
+    y_axis=1,
+    marker_overrides=None,
+    scatter_overrides=None,
+):
+    ''' Quick way to get a go.Scatter to put on your plot with some conveniences.
+
+    Args:
+        y_values: iterable of Y values can be a pandas Series or whatever
+        x_values: Optional, X values to go along with those. If not provided, y_values is expected to be a Series and
+            we'll use the index from it
+        dataset_name: Optional, name to put in the chart legend. If not provided, y_values is expected to be a Series
+            and we'll use its name
+        y_axis: Optional, Y-axis number. Defaults to the primary (1) Y axis. Note that axes 2-4 are used for RGB values.
+        marker_overrides: Optional, dict of go.Scatter.Marker attributes to override marker style.
+            Valid options are documented at https://plot.ly/python/reference/#scatter-marker
+        scatter_overrides: Optional, dict of go.Scatter attributes to override the Scatter.
+            Valid options are documented at https://plot.ly/python/reference/#scatter
+    '''
+    x_column_values = x_values or y_values.index
```

Comment:
Doesn't y_values have to be a dataframe for this to work? Description says "pandas Series or whatever"

Reply:
Series also have indexes. doc says `x_values: [...] If not provided, y_values is expected to be a Series and use the index from it`